boa/src/syntax/ast/node.rs

```diff
@@ -0,0 +1,300 @@
+use crate::syntax::ast::{
+    constant::Const,
+    op::{BinOp, Operator, UnaryOp},
+};
+use gc_derive::{Finalize, Trace};
+use std::{
+    collections::btree_map::BTreeMap,
+    fmt::{Display, Formatter, Result},
+};
+
+#[derive(Clone, Debug, Trace, Finalize, PartialEq)]
+/// A Javascript AST Node
+pub enum Node {
+    /// Create an array with items inside
+    ArrayDecl(Vec<Node>),
+    /// Create an arrow function with the given arguments and expression
+    ArrowFunctionDecl(Vec<FormalParameter>, Box<Node>),
+    /// Assign an expression to a value
+    Assign(Box<Node>, Box<Node>),
+    /// Run an operation between 2 expressions
+    BinOp(BinOp, Box<Node>, Box<Node>),
+    /// Run several expressions from top-to-bottom
+    Block(Vec<Node>),
+    /// Break statement with an optional label
+    Break(Option<String>),
+    /// Call a function with some values
+    Call(Box<Node>, Vec<Node>),
+    /// Conditional Operator ( ? : )
+    ConditionalOp(Box<Node>, Box<Node>, Box<Node>),
+    /// Make a constant value
+    Const(Const),
+    /// Const declaration
+    ConstDecl(Vec<(String, Node)>),
+    /// Construct an object from the function and arg{
+    Construct(Box<Node>, Vec<Node>),
+    /// Continue with an optional label
+    Continue(Option<String>),
+    /// Create a function with the given name, arguments, and expression
+    FunctionDecl(Option<String>, Vec<FormalParameter>, Box<Node>),
+    /// Gets the constant field of a value
+    GetConstField(Box<Node>, String),
+    /// Gets the [field] of a value
+    GetField(Box<Node>, Box<Node>),
+    /// [init], [cond], [step], body
+    ForLoop(
+        Option<Box<Node>>,
+        Option<Box<Node>>,
+        Option<Box<Node>>,
+        Box<Node>,
+    ),
+    /// Check if a conditional expression is true and run an expression if it is and another expression if it isn't
+    If(Box<Node>, Box<Node>, Option<Box<Node>>),
+    /// Let declaraton
+    LetDecl(Vec<(String, Option<Node>)>),
+    /// Load a reference to a value, or a function argument
+    Local(String),
+    /// New
+    New(Box<Node>),
+    /// Create an object out of the binary tree given
+    ObjectDecl(Box<BTreeMap<String, Node>>),
+    /// Object Declaration (replaces ObjectDecl)
+    Object(Vec<PropertyDefinition>),
+    /// Return the expression from a function
+    Return(Option<Box<Node>>),
+    /// Run blocks whose cases match the expression
+    Switch(Box<Node>, Vec<(Node, Vec<Node>)>, Option<Box<Node>>),
+    /// Spread operator
+    Spread(Box<Node>),
+    // Similar to Block but without the braces
+    StatementList(Vec<Node>),
+    /// Throw a value
+    Throw(Box<Node>),
+    /// Return a string representing the type of the given expression
+    TypeOf(Box<Node>),
+    /// Try / Catch
+    Try(
+        Box<Node>,
+        Option<Box<Node>>,
+        Option<Box<Node>>,
+        Option<Box<Node>>,
+    ),
+    This,
```

Comment:
Added documentation for `This`.
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/this
```suggestion
    /// A property of an execution context (global, function or eval) that, in non–strict mode, is always a reference to an object and in strict mode can be any value.
    This,
```

Reply:
I would add a small explanation and then the full explanation just below, since Rust documents this in one line, and the extra explanation in a spoiler:

```rust
    /// The `this` keyword.
    ///
    /// A property of an execution context (global, function or eval) that, in non–strict mode, is
    /// always a reference to an object and in strict mode can be any value.
```